docs/rules/html-self-closing.md

```diff
@@ -0,0 +1,78 @@
+---
+pageClass: "rule-details"
+sidebarDepth: 0
+title: "svelte/html-self-closing"
+description: "enforce self-closing style"
+---
+
+# svelte/html-self-closing
+
+> enforce self-closing style
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
+- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
+
+## :book: Rule Details
+
+You can choose either two styles for elements without content
+
+- always: `<div />`
+- never: `<div></div>`
+
+<ESLintCodeBlock fix>
+
+<!-- prettier-ignore-start -->
+<!--eslint-skip-->
+
+```svelte
+<script>
+  /* eslint svelte/html-self-closing: "error" */
+</script>
+
+<!-- ✓ GOOD -->
+<div />
+<p>Hello</p>
+<div><div /></div>
+<img>
+
+<!-- ✗ BAD -->
+<div></div>
+<p> </p>
+<div><div></div></div>
+<img />
+```
+
+<!-- prettier-ignore-end -->
+
+</ESLintCodeBlock>
+
+## :wrench: Options
+
+```json
+{
+  "svelte/html-self-closing": [
+    "error",
+    {
+      "html": {
+        "void": "always", // or "always" or "ignore"
+        "normal": "always", // or "never" or "ignore"
+        "component": "always" // or "never" or "ignore"
+      }
+    }
+  ]
+}
+```
+
+- `html.void` (`"always"` by default)... Style of HTML void elements
+- `html.component` (`"always"` by default)... Style of svelte components
+- `html.normal` (`"always"` by default)... Style of other elements
```

Comment:
> `<svelte:self></svelte:self>` seems to be affected by the `normal` option, but I think they need to share the `component` option or have different option.

https://github.com/ota-meshi/eslint-plugin-svelte/pull/186#discussion_r926286056

Reply:
@ota-meshi I'm having problem with implementing this. I've made all necessary changes, but with the selector `SvelteElement` the rule logic isn't being fired for some reason, but in the AST playground the `<svelte:head>` element has type`SvelteElement`
![image](https://user-images.githubusercontent.com/73162071/180601998-e42c4e7a-ea4a-4fe9-b9a8-52c1ae22a191.png)

Reply:
I think we can use `node.kind==="special"` to determine that.

Reply:
> I think we can use `node.kind==="special"` to determine that.

I did add that, but I had problems with the tests, now it's fixed